modules/core/src/smithy4s/Hints.scala

```diff
@@ -67,6 +67,11 @@ object Hints {
     }
     override def toString(): String =
       s"Hints(${all.mkString(", ")})"
+
+    override def hashCode(): Int = toMap.hashCode()
+    override def equals(obj: Any): Boolean = {
```

Comment:
created proper hashChode and equality methods which were desperately missing on the Hints construct.